### PR TITLE
refactor(avoidance_by_lane_change): update execution condition

### PR DESCRIPTION
**Pull request by zulfaqar-azmi-t4**
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 2147884</samp>

This pull request improves the lane change module for path avoidance by simplifying the logic, adding more flexibility, and enhancing the logging. It refactors the `specialRequiredCheck` function in `scene.cpp`, extends the `lane_change::utils` functions in `utils.cpp`, adds a new `processOnEntry` function in `interface.hpp` and `interface.cpp`, and modifies the logger name in `base_class.hpp`.

## Related links

None

## Tests performed

PSIM and evaluator

## Notes for reviewers

None

## Interface changes

Avoidance by lane change has higher priorities, so if previously, in the case where obstacle + lane change module is activated, now avoidance by lane change will be activated instead.

## Effects on system behavior

Avoidance by lane change has higher priorities, so if previously, in the case where obstacle + lane change module is activated, now avoidance by lane change will be activated instead.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_avoidance_module/include/behavior_path_avoidance_module/helper.hpp [hunk 1]
```diff
@@ -89,6 +89,7 @@ class AvoidanceHelper
   double getAvoidanceEgoSpeed() const
   {
     const auto & values = parameters_->velocity_map;
+
```
Comment (satoshi-ota) [inline planning/behavior_path_avoidance_module/include/behavior_path_avoidance_module/helper.hpp:92]: unnecessary change...?

File: planning/behavior_path_avoidance_module/include/behavior_path_avoidance_module/parameter_helper.hpp [hunk 2]
```diff
@@ -0,0 +1,364 @@
+// Copyright 2021 Tier IV, Inc.
```
Comment (satoshi-ota) [inline planning/behavior_path_avoidance_module/include/behavior_path_avoidance_module/parameter_helper.hpp:1]:
```suggestion
// Copyright 2023 TIER IV, Inc.
```

File: planning/behavior_path_avoidance_module/include/behavior_path_avoidance_module/helper.hpp (hunk 1 above)
Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_avoidance_module/include/behavior_path_avoidance_module/helper.hpp]: fixed in https://github.com/autowarefoundation/autoware.universe/pull/5869/commits/e30a50e04751ccbafaca49cc593609877654cfaa

**Review by satoshi-ota — APPROVED** [review]
LGTM :+1: I confirmed that normal LC was kicked and AbLC was launched instead.

https://github.com/autowarefoundation/autoware.universe/assets/44889564/0517a8e7-a51e-4df6-a6a6-08fe00c6e698

**Pull request merged**